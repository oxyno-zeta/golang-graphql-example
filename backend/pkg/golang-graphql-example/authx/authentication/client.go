package authentication

import (
	"net/url"
	"regexp"

	"github.com/gin-gonic/gin"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/config"
)

//go:generate mockgen -destination=./mocks/mock_Client.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/authx/authentication Client
type Client interface {
	// Middleware will redirect authentication to basic auth or OIDC depending on request path and resources declared.
	Middleware(unauthorizedPathRegexList []*regexp.Regexp) gin.HandlerFunc
	// OIDCEndpoints will set OpenID Connect endpoints for authentication and callback.
	OIDCEndpoints(router gin.IRouter) error
}

type providerEndpointsClaims struct {
	EndSessionEndpointURL *url.URL
	EndSessionEndpoint    string `json:"end_session_endpoint"`
}

func NewService(cfgManager config.Manager) Client {
	return &service{
		cfgManager: cfgManager,
	}
}
