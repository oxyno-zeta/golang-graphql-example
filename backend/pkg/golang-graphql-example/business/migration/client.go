package migration

import "github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database"

// Client will represent the service that will migrate database.
type Client interface {
	// Migrate.
	Migrate() error
}

func New(dbSvc database.DB) Client {
	return &service{dbSvc: dbSvc}
}
