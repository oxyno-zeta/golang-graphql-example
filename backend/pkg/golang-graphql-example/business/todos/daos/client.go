package daos

import (
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/business/todos/models"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database"
	"github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/database/pagination"
)

//go:generate mockgen -destination=./mocks/mock_Doa.go -package=mocks github.com/oxyno-zeta/golang-graphql-example/pkg/golang-graphql-example/business/todos/daos Dao
type Dao interface {
	MigrateDB() error
	GetAllPaginated(
		page *pagination.PageInput,
		sort *models.SortOrder,
		filter *models.Filter,
		projection *models.Projection,
	) ([]*models.Todo, *pagination.PageOutput, error)
	CreateOrUpdate(tt *models.Todo) (*models.Todo, error)
	FindByID(id string, projection *models.Projection) (*models.Todo, error)
}

func NewDao(db database.DB) Dao {
	return &dao{db: db}
}
