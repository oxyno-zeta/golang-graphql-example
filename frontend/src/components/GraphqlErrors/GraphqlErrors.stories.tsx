import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { ApolloError } from '@apollo/client';
import GraphqlErrors, { Props } from './GraphqlErrors';
import {
  forbiddenNetworkError,
  simpleForbiddenGraphqlError,
  simpleGraphqlErrorWithoutExtension,
  simpleInternalServerErrorGraphqlError,
} from './GraphqlErrors.storage-test';

export default {
  title: 'Components/GraphqlErrors',
  component: GraphqlErrors,
} as ComponentMeta<typeof GraphqlErrors>;

const Template: ComponentStory<typeof GraphqlErrors> = function C(args: Props) {
  return <GraphqlErrors {...args} />;
};

export const Playground = Template.bind({});
Playground.args = {
  error: new ApolloError({
    errorMessage: 'network apollo error',
    networkError: forbiddenNetworkError,
  }),
};

export const NetworkError: ComponentStory<typeof GraphqlErrors> = function C() {
  return (
    <GraphqlErrors
      error={
        new ApolloError({
          errorMessage: 'network apollo error',
          networkError: forbiddenNetworkError,
        })
      }
    />
  );
};

export const OneGraphQLErrorWithoutExtension: ComponentStory<typeof GraphqlErrors> = function C() {
  return (
    <GraphqlErrors
      error={
        new ApolloError({
          errorMessage: 'one graphql apollo error',
          graphQLErrors: [simpleGraphqlErrorWithoutExtension],
        })
      }
    />
  );
};

export const OneGraphQLErrorWithExtension: ComponentStory<typeof GraphqlErrors> = function C() {
  return (
    <GraphqlErrors
      error={
        new ApolloError({
          errorMessage: 'one graphql apollo error',
          graphQLErrors: [simpleForbiddenGraphqlError],
        })
      }
    />
  );
};

export const TwoGraphQLErrorWithExtension: ComponentStory<typeof GraphqlErrors> = function C() {
  return (
    <GraphqlErrors
      error={
        new ApolloError({
          errorMessage: 'two graphql apollo error',
          graphQLErrors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError],
        })
      }
    />
  );
};

export const OneGraphQLErrorWithoutMargin: ComponentStory<typeof GraphqlErrors> = function C() {
  return (
    <GraphqlErrors
      noMargin
      error={
        new ApolloError({
          errorMessage: 'one graphql apollo error',
          graphQLErrors: [simpleForbiddenGraphqlError],
        })
      }
    />
  );
};
