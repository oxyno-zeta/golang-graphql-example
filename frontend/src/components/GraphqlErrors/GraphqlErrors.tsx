import React, { Key, ReactNode } from 'react';
import Typography, { TypographyProps } from '@mui/material/Typography';
import Box from '@mui/material//Box';
import { ApolloError, ServerError } from '@apollo/client';
import { useTranslation } from 'react-i18next';
import type { SxProps } from '@mui/material';

export interface Props {
  error?: ApolloError;
  errors?: ApolloError[];
  noMargin?: boolean;
  containerBoxSx?: SxProps;
  errorTitleTypographyProps?: TypographyProps;
  errorElementTypographyProps?: TypographyProps;
  ulSx?: SxProps;
  liSx?: SxProps;
}

export interface CustomNetworkError {
  message: string;
  path: null | undefined | ReadonlyArray<string | number>;
}

const defaultProps = {
  error: null,
  errors: [],
  noMargin: false,
  containerBoxSx: {},
  errorTitleTypographyProps: {},
  errorElementTypographyProps: {},
  ulSx: {},
  liSx: {},
};

/* eslint-disable react/no-array-index-key */
function GraphqlErrors({
  error,
  errors,
  noMargin,
  containerBoxSx,
  errorTitleTypographyProps,
  errorElementTypographyProps,
  ulSx,
  liSx,
}: Props) {
  // Initialize translate
  const { t } = useTranslation();

  // Check if error or errors are set
  if ((!error || !errors) && errors && errors.length === 0) {
    // Display nothing
    return null;
  }

  // Build error list
  let errorList = errors;
  // Check if error is set
  if (error) {
    // Override array
    errorList = [error];
  }

  const elements: ReactNode[] = [];

  // Loop over errors
  errorList?.forEach((err, mainIndex) => {
    const contents: { content: string; key: Key }[] = [];
    if (
      err.networkError &&
      (!(err.networkError as ServerError).result ||
        Object.keys((err.networkError as ServerError).result).length === 0 ||
        !err.graphQLErrors)
    ) {
      contents.push({ content: err.networkError.message, key: mainIndex });
    }

    if (err.graphQLErrors) {
      contents.push(
        ...err.graphQLErrors.map(({ message, extensions }, i) => {
          let mess = message;
          // Check if there is a code in extensions
          if (extensions && extensions.code) {
            mess = t(`common.errorCode.${extensions.code}`);
          }

          return { content: mess, key: `${mainIndex}-graphQLErrors-${i}` };
        }),
      );
    }

    if (
      err.networkError &&
      (err.networkError as ServerError).result &&
      (err.networkError as ServerError).result.errors
    ) {
      contents.push(
        ...((err.networkError as ServerError).result.errors as [CustomNetworkError]).map(({ message, path }, i) => ({
          content: `${path?.join('.')} ${message}`,
          key: `${mainIndex}-networkError-${i}`,
        })),
      );
    }

    // Check if a content have been detected
    if (contents) {
      // Save element
      elements.push(
        ...contents.map(({ content, key }) => (
          <Box component="li" key={key} sx={liSx}>
            <Typography color="error" {...errorElementTypographyProps}>
              {content}
            </Typography>
          </Box>
        )),
      );
    }
  });

  return (
    <Box sx={{ color: 'error', margin: noMargin ? 0 : 8, ...containerBoxSx }}>
      <Typography color="error" {...errorTitleTypographyProps}>
        {t('common.errors')}:
      </Typography>
      <Box component="ul" sx={ulSx}>
        {elements}
      </Box>
    </Box>
  );
}
/* eslint-enable */

GraphqlErrors.defaultProps = defaultProps;

export default GraphqlErrors;
