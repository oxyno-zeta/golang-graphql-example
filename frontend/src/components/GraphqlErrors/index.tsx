import GraphqlErrors, { Props } from './GraphqlErrors';

export default GraphqlErrors;
export type { Props };
