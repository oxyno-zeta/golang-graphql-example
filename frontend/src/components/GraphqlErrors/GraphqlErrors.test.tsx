import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloError, ServerError } from '@apollo/client';
import { GraphQLError } from 'graphql';

import GraphqlErrors from './GraphqlErrors';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

// Build data for tests
export const forbiddenNetworkError: ServerError = {
  name: 'Forbidden',
  message: 'Forbidden',
  response: new Response(),
  statusCode: 403,
  result: {},
};
export const forbiddenNetworkErrorWithErrors: ServerError = {
  name: 'Forbidden',
  message: 'Forbidden',
  response: new Response(),
  statusCode: 403,
  result: { errors: [{ message: 'fake message', path: ['fake', 'path'] }] },
};
export const forbiddenNetworkErrorWithMultipleErrors: ServerError = {
  name: 'Forbidden',
  message: 'Forbidden',
  response: new Response(),
  statusCode: 403,
  result: {
    errors: [
      { message: 'fake message', path: ['fake', 'path'] },
      { message: 'fake message 2', path: ['fake', 'path2'] },
    ],
  },
};
export const simpleGraphqlErrorWithoutExtension: GraphQLError = new GraphQLError('simple graphql error');
export const simpleForbiddenGraphqlError: GraphQLError = new GraphQLError('forbidden graphql error', {
  extensions: { code: 'FORBIDDEN' },
});
export const simpleInternalServerErrorGraphqlError: GraphQLError = new GraphQLError(
  'internal server error graphql error',
  {
    extensions: { code: 'INTERNAL_SERVER_ERROR' },
  },
);

describe('GraphqlErrors', () => {
  it('should return null when no error or errors are present', () => {
    const component = renderer.create(<GraphqlErrors />);
    const tree = component.toJSON();

    expect(tree).toBeNull();
  });

  it('should display a network error when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'network apollo error',
            networkError: forbiddenNetworkError,
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('Forbidden');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a network error with 1 error when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'network apollo error',
            networkError: forbiddenNetworkErrorWithErrors,
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('fake.path fake message');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a network error with multiple errors when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'network apollo error',
            networkError: forbiddenNetworkErrorWithMultipleErrors,
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(3);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('fake.path fake message');
    expect(allP[2].props.children).toEqual('fake.path2 fake message 2');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a network error when errors are present', () => {
    const component = renderer.create(
      <GraphqlErrors
        errors={[
          new ApolloError({
            errorMessage: 'network apollo error',
            networkError: forbiddenNetworkError,
          }),
        ]}
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('Forbidden');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a graphql error without extension when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'one graphql apollo error',
            graphQLErrors: [simpleGraphqlErrorWithoutExtension],
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('simple graphql error');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a graphql error without extension when errors are present', () => {
    const component = renderer.create(
      <GraphqlErrors
        errors={[
          new ApolloError({
            errorMessage: 'one graphql apollo error',
            graphQLErrors: [simpleGraphqlErrorWithoutExtension],
          }),
        ]}
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('simple graphql error');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a graphql error with extension when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'one graphql apollo error',
            graphQLErrors: [simpleForbiddenGraphqlError],
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('common.errorCode.FORBIDDEN');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display a graphql error with extension when errors are present', () => {
    const component = renderer.create(
      <GraphqlErrors
        errors={[
          new ApolloError({
            errorMessage: 'one graphql apollo error',
            graphQLErrors: [simpleForbiddenGraphqlError],
          }),
        ]}
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(2);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('common.errorCode.FORBIDDEN');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display two graphql error with extension when error is present', () => {
    const component = renderer.create(
      <GraphqlErrors
        error={
          new ApolloError({
            errorMessage: 'two graphql apollo error',
            graphQLErrors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError],
          })
        }
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(3);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('common.errorCode.FORBIDDEN');
    expect(allP[2].props.children).toEqual('common.errorCode.INTERNAL_SERVER_ERROR');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display two graphql error with extension when errors are present (1 item with 2)', () => {
    const component = renderer.create(
      <GraphqlErrors
        errors={[
          new ApolloError({
            errorMessage: 'two graphql apollo error',
            graphQLErrors: [simpleForbiddenGraphqlError, simpleInternalServerErrorGraphqlError],
          }),
        ]}
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(3);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('common.errorCode.FORBIDDEN');
    expect(allP[2].props.children).toEqual('common.errorCode.INTERNAL_SERVER_ERROR');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });

  it('should display two graphql error with extension when errors are present (2 items)', () => {
    const component = renderer.create(
      <GraphqlErrors
        errors={[
          new ApolloError({
            errorMessage: 'two graphql apollo error',
            graphQLErrors: [simpleForbiddenGraphqlError],
          }),
          new ApolloError({
            errorMessage: 'two graphql apollo error',
            graphQLErrors: [simpleInternalServerErrorGraphqlError],
          }),
        ]}
      />,
    );
    const tree = component.toJSON();

    const allP = component.root.findAllByType('p');
    expect(allP).toHaveLength(3);
    expect(allP[0].props.children).toEqual(['common.errors', ':']);
    expect(allP[1].props.children).toEqual('common.errorCode.FORBIDDEN');
    expect(allP[2].props.children).toEqual('common.errorCode.INTERNAL_SERVER_ERROR');

    expect(tree).not.toBeNull();
    expect(tree).toMatchSnapshot();
  });
});
