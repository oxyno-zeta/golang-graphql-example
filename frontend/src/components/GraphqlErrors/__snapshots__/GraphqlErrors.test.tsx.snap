// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphqlErrors should display a graphql error with extension when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a graphql error with extension when errors are present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a graphql error without extension when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          simple graphql error
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a graphql error without extension when errors are present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          simple graphql error
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a network error when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          Forbidden
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a network error when errors are present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          Forbidden
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a network error with 1 error when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          fake.path fake message
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display a network error with multiple errors when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          fake.path fake message
        </p>
      </li>
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          fake.path2 fake message 2
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display two graphql error with extension when error is present 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display two graphql error with extension when errors are present (1 item with 2) 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>,
]
`;

exports[`GraphqlErrors should display two graphql error with extension when errors are present (2 items) 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-z0zjqa{margin:64px;}.css-z0zjqa color{main:#d32f2f;light:#ef5350;dark:#c62828;contrast-text:#fff;}",
      }
    }
    data-emotion="css z0zjqa"
  />,
  <div
    className="MuiBox-root css-z0zjqa"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1nkhxq8-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;color:#d32f2f;}",
        }
      }
      data-emotion="css 1nkhxq8-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
    >
      common.errors
      :
    </p>
    <ul
      className="MuiBox-root css-0"
    >
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.FORBIDDEN
        </p>
      </li>
      <li
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
        >
          common.errorCode.INTERNAL_SERVER_ERROR
        </p>
      </li>
    </ul>
  </div>,
]
`;
